[docs] Update instructions for yarn modern (#44890)
expo/expo: docs/pages/more/create-expo.mdx

#### EAS installation

> Yarn Modern on EAS requires enabling [Corepack](https://nodejs.org/api/corepack.html) for the build. Set [`corepack`](/eas/json/#corepack) to `true` in your build profile in **eas.json**:

Reviewer: ```suggestion
Yarn Modern on EAS requires enabling [Corepack](https://github.com/nodejs/corepack) for the build. Set [`corepack`](/eas/json/#corepack) to `true` in your build profile in **eas.json**:
```

The reason for this is that Corepack has been ejected from Node - it's non standard and the only thing it shows on the original link is:

<img width="1011" height="271" alt="Image" src="https://github.com/user-attachments/assets/88d1adf7-b2c7-4281-94ae-0d70f35a832c" />

```json eas.json
{
  "build": {
    "production": {
      "corepack": true
    }
  }
}
```

> Then pin the Yarn version in your project's **package.json** using the [`packageManager`](https://nodejs.org/api/packages.html#packagemanager) field. Running `yarn set version <version>` locally will update this field for you:

Reviewer 2: ```suggestion
Then, pin the Yarn version in your project's **package.json** using the [`packageManager`](https://nodejs.org/api/packages.html#packagemanager) field. Running `yarn set version <version>` locally will update this field for you:
```

```json package.json
{
  "packageManager": "yarn@4.14.1"
}
```

> With these in place, Corepack automatically downloads and uses the pinned Yarn version when EAS installs dependencies.

Reviewer 2: ```suggestion
After adding both of the above configurations, Corepack automatically downloads and uses the pinned Yarn version when EAS installs dependencies.
```

Small nit. I thi m we should call out things for that they are, for example, in this case, the above both configuration blocks are
